pkg/operators/limiter/limiter.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package limiter is a data operator that operates on array data sources by
+// limiting the number of entries in each batch of data. A great scenario for
+// this operator is when you are already sorting data within an array of data
+// and you want to filter out the top `X` entries.
+package limiter
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	name            = "limiter"
+	ParamMaxEntries = "max-entries"
```

Comment:
Maybe we can use `--limit` now as well? This would again move things more into an SQL direction that people might be more familiar with. But I don't have a super strong opinion on it.

Reply:
Let's see what others think about it.

Reply:
/cc @mauriciovasquezbernal @alban

Reply:
I feel "max-entries" is easier to understand, but not a hard opinions as well.